src/DefinitionProvider.php

```diff
@@ -34,6 +35,7 @@ public function __construct(
         ?PropertyTypeResolver        $propertyTypeResolver = null,
         bool                         $serializePublicMethods = true,
         ?ConstructorResolver         $constructorResolver = null,
+        bool                         $omitNullValuesOnSerialization = false,
```

Comment:
I'm not sure if I like treating the `DefinitionProvider` as a "configuration bag" and can certainly move things around - an alternative that still keeps BC would be an additional parameter for the code generator and the mapper using reflection.

Reply:
Hmmm, I think your hunch is right. Let's move it to the other places.